pkg/tfbridge/config_encoding.go

```diff
@@ -204,9 +181,15 @@ func (enc *configEncoding) UnmarshalProperties(props *structpb.Struct,
 }
 
 // Inverse of UnmarshalProperties, with additional support for top-level (but not nested) secrets.
-func (enc *configEncoding) MarshalProperties(props resource.PropertyMap,
-	opts plugin.MarshalOptions) (*structpb.Struct, error) {
-	opts.KeepSecrets = true
+func (enc *ConfigEncoding) MarshalProperties(props resource.PropertyMap) (*structpb.Struct, error) {
+	opts := plugin.MarshalOptions{
```

Comment:
It wasn't really configurable; this encoding implies a certain parameter set, inline it.